src/attributes/classes.js

```diff
@@ -24,19 +22,17 @@ jQuery.fn.extend({
 			});
 		}
 
-		if ( proceed ) {
-			// The disjunction here is for better compressibility (see removeClass)
-			classes = ( value || "" ).match( rnotwhite ) || [];
+		if ( typeof value === "string" && value ) {
+			classes = value.match( rnotwhite ) || [];
 
-			for ( ; i < len; i++ ) {
-				elem = this[ i ];
+			while ( ( elem = this[ i++ ] ) ) {
```

Comment:
So we would consistently use `while` everywhere? The reason to do this is to make code smaller?

>  I'm suggesting that we do so anyway (as is being done here) and document that such collections are invalid.

I'm a bit scare of that the less user complains the better, but that is not a strong position.